src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -214,6 +214,7 @@ private IServiceCollection BuildCommonServices()
 
             services.AddTransient<IApplicationBuilderFactory, ApplicationBuilderFactory>();
             services.AddTransient<IHttpContextFactory, HttpContextFactory>();
+            services.AddScoped<IMiddlewareFactory, MiddlewareFactory>();
```

Comment:
Right. To avoid passing the `IServiceProvider` as part of the interface contract the default factory itself is scoped.